### DISCUSSION TITLE
Lifetime parameters or bounds on method `create` do not match the trait declaration

### DISCUSSION THREAD

**Discussion by phdavis1027** [bikeshedder/deadpool#303] [Q&A] [answered]
Hi all,

Preface: I have a lot of amateur Rust experience, but there are many areas I'm still rather fuzzy on,

The title is the compiler error I get from the following code: 

```rust
    async fn create(& self) -> Result<Self::Type, Self::Error> {
        let conn = Connection::new(
            (self.host.as_str(), self.port),
            &self.account,
            &self.conn_config,
        )?;
        SyncWrapper::new(deadpool_runtime::Runtime::Tokio1, move|| conn.login()).await 
    }
```

I am confused because there are no lifetime bounds on the trait method as far as I see. I even `gd`'d with the my LSP and the local copy I have doesn't see any trait bounds. My code is modeled on the sqlite example.  Any help would be appreciated.

**Comment by bikeshedder** [comment, marked as answer]
Deadpool 0.10 from crates.io still uses the  `async-trait` crate. The upcoming release will drop this and use async traits as introduced in rust 1.75.

I guess you have used the code from the GitHub repository and used the release from crates.io? That would explain the error message you are getting.

To fix that you can either add the async-trait crate to your own app or use the deadpool version from this repository.

See:

- #298

**Comment by phdavis1027** [comment]
No, I just ran `cargo add deadpool`. The re-exported `async_trait` fixed my problem though, thanks.
